src/Api/HL/Search.php

```diff
@@ -108,15 +109,16 @@ private function getSelectCriteriaForProperty(string $prop_name, bool $distinct_
                 if (str_contains($sql_field, '.')) {
                     $join_name = explode('.', $sql_field, 2)[0];
                     if (array_key_exists($join_name, $this->joins) && $this->joins[$join_name]['parent_type'] === 'array') {
-                        $expression = 'IFNULL(' . $expression . ', 0x0)';
+                        $expression = QueryFunction::ifnull($sql_field, new QueryExpression($DB::quoteValue('0x0')));
```

Comment:
If you surround the unicode char representation by quotes, you will probably optain a `0x0` string instead of a `null` value.
```suggestion
                        $expression = QueryFunction::ifnull($sql_field, new QueryExpression('0x0'));
```